### PR TITLE
POPOVER

### PR DESCRIPTION
**Pull request by michsa**
NEW POPOVER:

- takes component props `Trigger` and `Popup` to control the anchor/trigger component and popup container respectively (the trigger component *must* handle refs for this popover to work)
- takes prop props `triggerProps` and `popupProps` to pass props to those component props (mostly as a matter of convenience on account of the refs thing)
- takes positioning props `placement` (top/bottom/left/right) and `side` (left/right - should probably rename these) to control which side of the anchor element the popup goes on
- continues to take state props `open`, `isOpen`, and `onChange` (🚨 **BREAKING:** 🚨 formerly `onClose`)
- takes a `label` prop for the children of the trigger, and passes `children` to the children of the popup

NEW DROPDOWN/DIALOG:

- added Dialog, which is identical to Dropdown except for the padding on its popup container
- Dropdown's API is now different from Popover's: Dropdown (and the new Dialog) come bundled with a trigger button that they can pass props into
- does *not* support `Trigger`, `triggerProps`, `Popup`, or `popupProps`
- *does* support `trigger` (one of "button" or "icon"), which controls what kind of button the trigger uses (Button or IconButton)
- also supports `label` and `children` for the children of the trigger and popup, respectively
- passes extra props through to the trigger component, so it supports any props of Button or IconButton (e.g. `icon`)

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import _ from 'lodash/fp'
 import F from 'futil'
 import { useState } from 'react'
 import { mapProps } from 'recompose'
+import { loremIpsum } from 'lorem-ipsum'
```
Comment (daedalus28) [inline src/utils.js]: this is in src/utils, but `lorem-ipsum` should not be bundled with grey-vest

File: src/Box.js [hunk 2]
```diff
@@ -1,23 +1,30 @@
...
       boxShadow: getVariants(props, theme.boxShadows, 'normal'),
-      padding: _.flow(
-        F.flowMap(theme.space, F.append('px')),
-        _.join(' ')
-      )([coalesce([paddingY, padding]), coalesce([paddingX, padding])]),
+      ..._.flow(
+        _.map(theme.space),
+        _.apply(pad)
+      )([
+        paddingY || _.head(padding) || padding,
```
Comment (daedalus28) [inline src/Box.js]: this should probably use coalesce because `0` is a valid value

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>:exclamation: This PR is BIG (+370 -267)  
Please keep it below 500 net changes</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>The README has not been updated. Please update the README.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Could not find any browser results.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 644fc32635c50a8ab4dffb0c39a505fae5b2adaf
</p>

File: src/utils.js (hunk 1 above)
Comment (michsa) [reply src/utils.js]: moved to stories/utils

**Pull request merged**